cmd/insprd/memory/brokers/manager.go

```diff
@@ -28,3 +28,9 @@ func GetBrokerMemory() Manager {
 	}
 	return brokerMemory
 }
+
+// ResetBrokerMemory makes the BrokersManagers singleton points to nil,
+// used only in tests
+func ResetBrokerMemory() {
+	brokerMemory = nil
+}
```

Comment:
I think this is a pore practice for singletons, effectively it goes against the singleton design. If it is necessary to test you could use this structure's mock.

Reply:
If it doesn't work with mock leave it